scalafix-interfaces/src/main/java/scalafix/interfaces/ScalafixOutput.java

```diff
@@ -0,0 +1,66 @@
+package scalafix.interfaces;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+public interface ScalafixOutput {
+
+    /**
+     *
+     * @return original path of the file to be fixed
+     */
+    Path getPath();
+
+    /**
+     *
+     * @return corresponding list of rules that has been evaluated
+     */
+    ScalafixRule[] getRules();
+
+    /**
+     *
+     * @return boolean true if there are no error when scalafix run on the file
+     */
+    boolean isSuccessful();
+
+    ScalafixError[] getErrors();
+
+    /**
+     *
+     * @return list of diagnostics
+     */
+    ScalafixDiagnostic[] getDiagnostics();
+
+    /**
+     *
+     * @return if the fixed file is different from the input file, this method return non empty unified diff
+     */
+    Optional<String> getUnifiedDiff();
+
+    ScalafixPatch[] getPatches();
+
+    /**
+     *
+     * @return the output content after evaluating scalafix if no error. The original file will stay unchanged.
+     */
+    Optional<String> getOutputFixed();
```

Comment:
`output` was not really clear to me since it's the name of the class itself, while I think here it means the content of the path after the fix is applied. Also, the fix does not really show that this is a dry-run version of `applyPatches`. What about
```
interface ScalafixTargetEvaluation {
  Path getTarget(); 
  Optional<String> previewPatches();
```

(we should stick to one wording for file/target/path to designate what scalafix runs on by the way - not sure what's the best or what's already used)